subsys/net/ip/ipv6.c

```diff
@@ -577,10 +580,6 @@ enum net_verdict net_ipv6_input(struct net_pkt *pkt, bool is_loopback)
 	}
 
 	if (verdict == NET_DROP) {
-		if (nexthdr == IPPROTO_ICMPV6) {
```

Comment:
Ah same issue I had in ipv4, and of course I did not check is I did the same mistake in ipv6.

Please fix it the same way as in ipv4.c:242